tunnel/intra/doh/doh.go

```diff
@@ -53,16 +53,21 @@ const (
 
 // Summary is a summary of a DNS transaction, reported when it is complete.
 type Summary struct {
-	Latency  float64 // Response (or failure) latency in seconds
-	Query    []byte
-	Response []byte
-	Server   string
-	Status   int
+	Latency    float64 // Response (or failure) latency in seconds
+	Query      []byte
+	Response   []byte
+	Server     string
+	Status     int
+	HTTPStatus int // Zero unless Status is Complete or HTTPError
 }
 
+// A Token is an opaque handle used to match responses to queries.
+type Token interface{}
```

Comment:
What fields/methods will the Java interface have to match responses to queries? It may be useful to document and enforce them here. Could we use the query ID?

Reply:
I realized that this is hard to understand in isolation, so I added a link to the corresponding code on the Java side.

I would be concerned about using the query ID because different clients could produce colliding IDs.  Also, IDs in general are a form of manual reference counting, with the potential for a leak if an ID is ever registered and then not cleaned up.  The token-passing design avoids the need for a map from IDs to state, and lets the garbage collector control object lifetimes.

Reply:
I see - the approach sounds reasonable to me.